src/backend/src/services/tasks.services.ts

```diff
@@ -258,4 +265,69 @@ export default class TasksService {
 
     return deletedTask.taskId;
   }
+
+  /**
+   * Sends the task deadline slack notifications for all tasks with a deadline of the given date
+   * @param deadline the deadline date as a number
+   */
+  static async sendTaskDeadlineSlackNotifications(deadline: number) {
+    const startOfDay = new Date(deadline);
+    const endOfDay = new Date(startOfDay);
+    endOfDay.setDate(startOfDay.getDate() + 1);
+
+    const tasks = await prisma.task.findMany({
+      where: {
+        deadline: {
+          gte: startOfDay,
+          lt: endOfDay
+        },
+        status: {
+          not: 'DONE'
+        }
+      },
+      include: {
+        assignees: {
+          include: {
+            userSecureSettings: true,
+            userSettings: true,
+            teamAsHead: true,
+            teamsAsLead: true,
+            teamsAsMember: true
+          }
+        },
+        wbsElement: {
+          include: {
+            project: { include: { teams: true } }
+          }
+        }
+      }
+    });
+
+    const teamTaskMap = new Map<string, TaskWithAssignees[]>();
+
+    tasks.forEach((task) => {
+      const teamSlackId = getTeamFromTaskAssignees(task.assignees);
```

Comment:
actually i just realized that currently tasks can only be assigned to the same team on the frontend since a project is "owned" by a team and the options to choose assignees will only show the applicable members of a team, so i don't think it is necessary to do that. then i can just pull the slackId from the project team

Reply:
nevermind, scratch that